reposilite-backend/src/main/kotlin/com/reposilite/token/AccessTokenFacade.kt

```diff
@@ -70,4 +82,49 @@ class AccessTokenFacade internal constructor(
     fun count(): Long =
         temporaryRepository.countAccessTokens() + persistentRepository.countAccessTokens()
 
+    fun getTokensResponse(manager: Boolean, tokens: Collection<AccessToken>): Result<TokensResponse, ErrorResponse> {
+        return manager.takeIf { it }
+            ?.let { TokensResponse(tokens.map { TokensResponse.AccessTokenField(it) }).asSuccess() } ?: errorResponse(
+            HttpCode.UNAUTHORIZED,
+            "You must be a manager to access this endpoint!"
+        )
+    }
+
+    fun getTokenResponse(
+        accepted: Boolean,
+        accessToken: AccessToken?
+    ): Result<TokensResponse.AccessTokenField, ErrorResponse> {
+        return accepted.takeIf { it }?.let {
+            accessToken?.let { TokensResponse.AccessTokenField(it).asSuccess() }
+                ?: errorResponse(HttpCode.BAD_REQUEST, "The token requested does not exist!")
+        } ?: errorResponse(
+            HttpCode.UNAUTHORIZED,
+            "You must be the owner of the token or a manager to access this endpoint!"
+        )
+    }
+
+    fun deleteTokenWithResponse(
+        manager: Boolean,
+        name: String
+    ): Result<TokensResponse.AccessTokenField, ErrorResponse> {
+        return manager.takeIf { it }?.let {
+            deleteToken(name)?.let { TokensResponse.AccessTokenField(it).asSuccess() }
+                ?: errorResponse(HttpCode.BAD_REQUEST, "Could not find a token to delete!")
+        } ?: errorResponse(HttpCode.UNAUTHORIZED, "You must be a manager to access this endpoint!")
+    }
+
+    fun createOrUpdateToken(
+        manager: Boolean,
+        request: CreateAccessTokenRequest
+    ): Result<CreateAccessTokenResponse, ErrorResponse> {
+        if (!manager) {
+            return errorResponse(HttpCode.UNAUTHORIZED, "You must be a manager to access this endpoint!")
+        }
+
+        if (getToken(request.name) != null) {
+            deleteToken(request.name)
+        }
+
+        return createAccessToken(request).asSuccess()
+    }
```

Comment:
A few notes related in general to these methods:

1. In general a facade should not be responsible for authorization, so rather than passing the `manager: Boolean` property, verify this in endpoints. 
2. `getTokenResponse` methods should be called more like just `getTokens`/`findTokens`
3. Passing `tokens: Collection<AccessToken>` looks kinda dirty, I think you can just take it from facade inside this metod